podpac/core/data/datasource.py

```diff
@@ -437,6 +465,9 @@ def eval(self, coordinates, output=None):
                     + "request Coordinates coordinate reference system ({})".format(coordinates.crs)
                 )
 
+        # get indexed boundary
+        self._requested_source_boundary = self.get_boundary(self._requested_source_coordinates_index)
+
```

Comment:
This call (and the get_boundary method) is only necessary for arrays of non-uniform boundaries. And since we aren't actually using the boundary anywhere in the interpolation, it isn't necessary at all. But I didn't want to forget about it. Otherwise, the datasource eval method is fundamentally unchanged.